src/plugins/intel_npu/src/plugin/npuw/accuracy/result.hpp

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/openvino.hpp"
+
+namespace ov {
+namespace npuw {
+namespace metrics {
+
+struct Result {
+    double metric = -1;
+    double threshold = -1;
+
+    operator bool() const {
+        // It is unexpected that calculated metric will be < 0.0,
+        // in that case, return that tensors are unequal.
+        return (metric <= threshold) && (metric >= 0.0);
+}
```

Comment:
That's a strange design choice tbh, but as it looks more like comparison result (a numeric value), not an accuracy check result (pass/fail), it should be the `Comparator::Result`.